frontend/app/.server/validation/email-address-schema.ts

```diff
@@ -0,0 +1,85 @@
+import isEmail from 'validator/es/lib/isEmail';
+import type { z } from 'zod';
+
+import { stringSchema } from '~/.server/validation/string-schema';
+
+/**
+ * Interface defining customizable error messages for email address validation schema.
+ */
+export interface EmailAddressSchemaErrorMessages {
+  /**
+   * Error message for when the email address format is invalid.
+   * @default 'Email address format is invalid.'
+   */
+  format_error?: string;
+
+  /**
+   * Error message for when the email address is not a string.
+   * @default 'Email address must be a string.'
+   */
+  invalid_type_error?: string;
+
+  /**
+   * Error message for when the email address exceeds the maximum length.
+   * @default 'Email address must be less than or equal to {maximum} characters
+   */
+  max_length_error?: string;
+
+  /**
+   * Error message for when the email address is required.
+   * @default 'Email address is required.'
+   */
+  required_error?: string;
+}
+
+/**
+ * Configuration options for email address validation, including maximum length and error messages.
+ */
+export interface EmailAddressSchemaOptions {
+  errorMessages?: EmailAddressSchemaErrorMessages;
+  maxLength?: number;
+}
+
+const DEFAULT_MESSAGES = {
+  format_error: 'Email address format is invalid.',
+  invalid_type_error: 'Email address must be a string.',
+  max_length_error: 'Email address must be less than or equal to {maximum} characters.',
+  required_error: 'Email address is required.',
+} as const satisfies Required<EmailAddressSchemaErrorMessages>;
```

Comment:
These should be i18n keys. Check out how Java does it: ><https://javaee.github.io/javaee-spec/javadocs/javax/validation/constraints/Email.html#message-->

Reply:
I disagree. Zod will always return issues with the path and code and default formatted message. It let  you override default error message like I did.

Reply:
We can say the same for Zod default messages. The purpose of this pull request is to add reusable zod custom schema. We can refine them in another PR, but don`t forget that email address schema message will be different based on his context. `Email address` vs `Confirm email address`. The base email validation schema is the same but the error messages will differ.

Reply:
I'm planning doing it in another PR. It defeats the purpose of this PR.

Reply:
I think that i18n error messages falls within the responsibility of this PR, but am tired of arguing so I'll merge it.